pkg/api/v1alpha1/cloudstackdatacenterconfig_webhook.go

```diff
@@ -87,7 +87,7 @@ func validateImmutableFieldsCloudStackCluster(new, old *CloudStackDatacenterConf
 	if old.Spec.Domain != new.Spec.Domain {
 		allErrs = append(
 			allErrs,
-			field.Invalid(field.NewPath("spec", "domain"), new.Spec.Domain, "field is immutable"),
+			field.Forbidden(field.NewPath("spec", "domain", new.Spec.Domain), "field is immutable"),
```

Comment:
What's the motivation for this change?

Reply:
CAPI uses forbidden for immutable fields, instead of Invalid. Invalid fields can generally mean its a string instead of int, but we want to forbid changing the value at all for immutable fields.

Reply:
Gotcha. In that case we should be good from Cloudstack, just need to resolve the conflicts